### PR TITLE
Clean Up

### PR DESCRIPTION
**Pull request by alcohol**
Needs discussion and feedback.

### CODE REVIEW

**Comment by alcohol** [comment]
I want to bring back `update.sh` and the templates, so ignore their removal for now.

File: README.md [hunk 1]
```diff
@@ -1,107 +1,113 @@
...
-    ```sh
-    $ sudo chmod +x /usr/local/bin/composer
-    ```
+## Suggestions
+
+We strive to deliver an image that is as lean as possible, aimed at running
+Composer only.
+
+Sometimes dependencies or [composer scripts] require the availability of certain
+PHP extensions. In this scenario, you have two options:
```
Comment (RobLoach) [inline README.md]: Could also be good to add some documentation on how to create your own `Dockerfile` to add some custom PHP extensions.

File: 1.2/Dockerfile [hunk 2]
```diff
@@ -1,11 +1,49 @@
...
-MAINTAINER Rob Loach <robloach@gmail.com>
+#
+# NOTE: THIS DOCKERFILE IS GENERATED VIA "update.sh"
+#
+# PLEASE DO NOT EDIT IT DIRECTLY.
+#
+FROM php:7-cli
 
-ENV COMPOSER_VERSION 1.2.0
+RUN DEBIAN_FRONTEND=noninteractive \
```
Comment (alcohol) [inline 1.2/Dockerfile]: You are correct, oversight on my part.

File: 1.2/Dockerfile [hunk 3]
```diff
@@ -1,11 +1,49 @@
...
+ && apt-get update \
+ && apt-get install -y curl git subversion openssh-client openssl mercurial \
+ && rm -r /var/lib/apt/lists/*
 
-# Install Composer
-RUN php /tmp/composer-setup.php --no-ansi --install-dir=/usr/local/bin --filename=composer --version=${COMPOSER_VERSION} && rm -rf /tmp/composer-setup.php
+ENV TINI_VERSION v0.10.0
+ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini /tini
+ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini.asc /tini.asc
```
Comment (alcohol) [inline 1.2/Dockerfile]: Alright, let me change that.

**Comment by alcohol** [comment]
I'm not certain having update.sh written in bash is the best idea. While it is definitely suitable enough for the task, it is less comprehensible than for example PHP, given our target audience. We might want to consider writing a script handling the template generation in PHP instead.

**Comment by alcohol** [comment]
Maybe this can be closed in favor of trying to get https://github.com/composer/docker in a shape where it can be submitted as an official image? We can then leave composer/composer around as legacy for the time being.

**Comment by sylus** [comment]
What is the issue with having update.sh in bash? This is exactly how the official repo's do it.

See: https://github.com/docker-library/mysql/

**Comment by alcohol** [comment]
@sylus pretty sure I included the reasoning in the very same paragraph. Not sure how much more descriptive you want it to be.

The script itself is nothing "official" at all. It is simply a helper.

**Comment by sylus** [comment]
We are talking about patterns and in most of the docker repo's they all have roughly the same script. I am merely stating we should follow how it is done in the repo's for consistency.

**Comment by sylus** [comment]
Aside from that I think the changes look good, I will try to test them out later in the week. Definitely would be nice to get this on its way to being official. Thanks all!

**Comment by alcohol** [comment]
I see. Their base images extend `FROM: composer/composer`. That is indeed something that must be taken into consideration.

I thought Docker best practices recommend you only extend from official images though?

**Comment by sylus** [comment]
I think it was under the assumption that composer was going to be an official image; if I am understanding correctly. 

However with the idea to keep the repository as a legacy one, shouldn't be too much of an issue :)

**Comment by sylus** [comment]
Hey also just wanted to add this related issue here, as they are considering adding composer directly as a script in docker-php. Not sure I agree but there are arguments for.

Here is the issue:

https://github.com/docker-library/php/pull/260

**Comment by RobLoach** [comment]
This isn't controversial, it's some valid clean up.

**Comment by RobLoach** [comment]
> Maybe this can be closed in favor of trying to get https://github.com/composer/docker

Seems like that is just this PR, without the git history?

**Comment by alcohol** [comment]
Yeah. It was a new repo, not a fork. If you object to losing the history though, I can force push this branch/pr to it.

**Comment by RobLoach** [comment]
We could just move the repo over to /composer instead.

**Comment by alcohol** [comment]
I'm not sure that would be a good idea. I don't know how the docker hub would deal with that. Nor am I in a position to orchestrate that. I would have to bug Jordi, but he isn't really into the docker stuff, so not sure when he would have time for that.

**Comment by alcohol** [comment]
https://hub.docker.com/_/composer/

**Pull request closed without merging**